app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -107,8 +108,10 @@ public void run() {
                     if (!registeredOrGotEventsThisRun) {
                         request.setProperty("dont_block", "true");
                     }
-                    JSONObject response = new JSONObject(request.execute("GET",
-                            "v1/events"));
+                    Response httpResponse =request.execute("GET",
+                            "v1/events");
+                    String json = httpResponse.body().string();
+                    JSONObject response = new JSONObject(json);
```

Comment:
may need to check the response code for success earlier, since it no longer throws an error.